### DISCUSSION TITLE
<pnp-collapsible> collapsed / not collapsed in if statement

### DISCUSSION THREAD

**Discussion by bckrma** [microsoft-search/pnp-modern-search#2783] [Ideas]
Hello,

is it possibile to use an {#if } statement to test if an <pnp-collapsible> is collapsed or not?
I am working on a result template to display some text within a <pnp-collapsible> component.
My goal is to display only a part of the text above the <pnp-collapsible> using the {{ellipsis}} helper, and remove this text when the <pnp-collapsible> is collapsed.

Is something like this possible?

Thank you in advance

**Comment by wobba** [comment]
Can you explain a bit more? When it's collapsed only the header shows right? So when do you want the ... to show or not?

**Reply by bckrma** [reply]
Sure, my goal is to display list elements with a long, multi-line text field, in a compact form.

Currently the list item looks like this in the search results webpart. Only a part of the multi-line text field is displayed.
![01](https://user-images.githubusercontent.com/48834093/220551749-be74ff6f-cb85-44fa-a0fe-5dbbaf4654c2.png)

The item looks like this when you click "Expand"
![02](https://user-images.githubusercontent.com/48834093/220552450-774029e0-0182-4195-ba51-696b5322746d.png)

What i would like to achieve is, that the upper text is hidden when you click on expand. 
But I am not sure if, or how this is possibile.

Thanks in advance!

**Reply by wobba** [reply]
Ah.. got it. I'm thinking it's not possible as the expand/collapse is not causing re-render/eval of the template as it's on page CSS toggling. Adding an extra section inside the component is likely the only way and support that it's hidden when expanded.

**Reply by bckrma** [reply]
okay, thank you very much for the clarification.
I think at this point i leave it as it is since my HTML/CSS knowledge is to little to accomplish this otherwise.
